app/views/facilities/_azlist.html.haml

```diff
@@ -0,0 +1,14 @@
+.azlist
+
+    - @azlist.to_a.map.with_index do | value, index|
+        - if value[1].empty?
+            %p= value[0]
+        - else
+            %a.azListing{:class=>get_Classname_For_Facility(index, value[0])} 
```

Comment:
The helper function is so that we can get the same value no matter where we check. We actually had a big problem working with this because if we simply set the class name to the letter, it breaks if the letter is non-alphanumeric. `class="!"` is valid syntax, but JQuery will not allow `$('.!')`, and CSS will not allow `.! { }`.

What we had to do is use a number that corresponds to the index in the object. This is why you will see the classes look like `class="azlist16"`. However, since that classname is given to elements in the AZ List partial as well as the index page, we need to be sure we are generating the same classname for each. @robstolarz originally suggested the use of a function for this, and we think it's the best solution. I would say a large chunk of our time spent developing this feature was getting the classnames correct.